cumulus_library/studies/core/reference_sql/builder_observation.sql

```diff
@@ -95,43 +95,167 @@ CREATE TABLE core__observation_dn_code AS (
+        FROM
+            observation AS t,
+            UNNEST(t."component") AS parent (r)
+    ),
+
+    system_dataabsentreason_0 AS (
+        SELECT DISTINCT
+            s.id AS id,
+            s.row,
+            u.coding.code,
+            u.coding.display,
+            u.coding.system,
+            u.coding.userSelected
+        FROM
+            flattened_rows AS s,
+            UNNEST(s.dataabsentreason.coding) AS u (coding)
+    ), --noqa: LT07
+
+    union_table AS (
+        SELECT
+            id,
+            row,
+            system,
+            code,
+            display,
+            userSelected
+        FROM system_dataabsentreason_0
+        
     )
-        AS t ("id","row","code","system","display","userSelected")
-    WHERE 1 = 0 -- ensure empty table
+    SELECT
+        id,
+        row,
+        code,
+        system,
+        display,
+        userSelected
+    FROM union_table
 );
 
+
 -- ###########################################################
 
-CREATE TABLE IF NOT EXISTS "cumulus_mhg_dev_db"."core__observation_component_interpretation"
-AS (
-    SELECT * FROM (
-        VALUES
-        (cast(NULL AS varchar),cast(NULL AS bigint),cast(NULL AS varchar),cast(NULL AS varchar),cast(NULL AS varchar),cast(NULL AS boolean))
+CREATE TABLE core__observation_component_interpretation AS (
+    WITH
+
+    flattened_rows AS (
+        SELECT DISTINCT
+            t.id AS id,
+            ROW_NUMBER() OVER (PARTITION BY id) AS row,
+            r."interpretation"
+        FROM
+            observation AS t,
+            UNNEST(t."component") AS parent (r)
+    ),
+
+    child_flattened_rows AS (
+        SELECT DISTINCT
+            s.id,
+            s.row, -- keep the parent row number
+            u."interpretation"
+        FROM
+            flattened_rows AS s,
+            UNNEST(s.interpretation) AS u ("interpretation")
+    ),
+
+    system_interpretation_0 AS (
+        SELECT DISTINCT
+            s.id AS id,
+            s.row,
+            u.coding.code,
+            u.coding.display,
+            u.coding.system,
+            u.coding.userSelected
+        FROM
+            child_flattened_rows AS s,
+            UNNEST(s.interpretation.coding) AS u (coding)
+    ), --noqa: LT07
+
+    union_table AS (
+        SELECT
+            id,
+            row,
+            system,
+            code,
+            display,
+            userSelected
+        FROM system_interpretation_0
+        
     )
-        AS t ("id","row","code","system","display","userSelected")
-    WHERE 1 = 0 -- ensure empty table
+    SELECT
+        id,
+        row,
+        code,
+        system,
+        display,
+        userSelected
+    FROM union_table
 );
 
+
 -- ###########################################################
 
-CREATE TABLE IF NOT EXISTS "cumulus_mhg_dev_db"."core__observation_component_valuecodeableconcept"
+CREATE TABLE IF NOT EXISTS "main"."core__observation_component_valuecodeableconcept"
```

Comment:
this is an aside, but if you're generating these against duckdb, and i'm generating them against athena, there is going to be constant churn here (though my dev db is definetely also the wrong target)

We should probably pick one target and both try to use that all the time - I was going to say making a staging version of the cerner DB, but that means waiting a half hour for a build to run. Perhaps a standalone for this purpose based on synthea data in athena?

Reply:
or, i mean, just always do it in duckdb, though that's not quite accurate as a reference for third parties as to what's going to run live

Reply:
I added a [section to the contributing docs](https://github.com/smart-on-fhir/cumulus-library/blob/main/CONTRIBUTING.md#rebuilding-the-reference-sql) a while back with a one-liner that pointed at the duckdb data. So that's our currently-documented process. I didn't mean to slide that by you, I just wanted to document a process that worked.

I agree it's not 100% accurate, but it's easy for randos if they ever filed a PR and quick for everyone. I guess I also have questions about the utility of being fully accurate, versus a smoke test catch of "oh shoot, my SQL is crazier than I thought" (which is how I've been using the reference SQL)

Reply:
ah yep, that's fine, now i just need to remember it